From e3c89409486ea9c518fff152944ec15c7831a6c7 Mon Sep 17 00:00:00 2001
From: Alexander Pivovarov <apivovarov@gmail.com>
Date: Wed, 13 May 2015 23:27:02 -0700
Subject: [PATCH 0174/1164] CDH-29061:HIVE-10706 Make
 vectorized_timestamp_funcs test more stable
 (Alexander Pivovarov, reviewed by Jason Dere)

---
 .../clientpositive/vectorized_timestamp_funcs.q    |   36 ++++-----
 .../spark/vectorized_timestamp_funcs.q.out         |   78 +++++++++----------
 .../tez/vectorized_timestamp_funcs.q.out           |   80 ++++++++++----------
 .../vectorized_timestamp_funcs.q.out               |   80 ++++++++++----------
 4 files changed, 137 insertions(+), 137 deletions(-)

diff --git a/ql/src/test/queries/clientpositive/vectorized_timestamp_funcs.q b/ql/src/test/queries/clientpositive/vectorized_timestamp_funcs.q
index 8a2d5aa..a6b7761 100644
--- a/ql/src/test/queries/clientpositive/vectorized_timestamp_funcs.q
+++ b/ql/src/test/queries/clientpositive/vectorized_timestamp_funcs.q
@@ -142,31 +142,31 @@ FROM alltypesorc_string;
 
 -- SUM of timestamps are not vectorized reduce-side because they produce a double instead of a long (HIVE-8211)...
 EXPLAIN SELECT
-  sum(ctimestamp1)
+  round(sum(ctimestamp1), 3)
 FROM alltypesorc_string;
 
 SELECT
- sum(ctimestamp1)
+ round(sum(ctimestamp1), 3)
 FROM alltypesorc_string;
 
 EXPLAIN SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string;
 
 SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string;
\ No newline at end of file
diff --git a/ql/src/test/results/clientpositive/spark/vectorized_timestamp_funcs.q.out b/ql/src/test/results/clientpositive/spark/vectorized_timestamp_funcs.q.out
index 85da980..bfdc2b6 100644
--- a/ql/src/test/results/clientpositive/spark/vectorized_timestamp_funcs.q.out
+++ b/ql/src/test/results/clientpositive/spark/vectorized_timestamp_funcs.q.out
@@ -701,12 +701,12 @@ POSTHOOK: Input: default@alltypesorc_string
 1969-12-31 23:59:44.088	1970-01-01 00:00:15.007	39	40
 PREHOOK: query: -- SUM of timestamps are not vectorized reduce-side because they produce a double instead of a long (HIVE-8211)...
 EXPLAIN SELECT
-  sum(ctimestamp1)
+  round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 POSTHOOK: query: -- SUM of timestamps are not vectorized reduce-side because they produce a double instead of a long (HIVE-8211)...
 EXPLAIN SELECT
-  sum(ctimestamp1)
+  round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 STAGE DEPENDENCIES:
@@ -746,7 +746,7 @@ STAGE PLANS:
                 outputColumnNames: _col0
                 Statistics: Num rows: 1 Data size: 8 Basic stats: COMPLETE Column stats: NONE
                 Select Operator
-                  expressions: _col0 (type: double)
+                  expressions: round(_col0, 3) (type: double)
                   outputColumnNames: _col0
                   Statistics: Num rows: 1 Data size: 8 Basic stats: COMPLETE Column stats: NONE
                   File Output Operator
@@ -765,38 +765,38 @@ STAGE PLANS:
         ListSink
 
 PREHOOK: query: SELECT
- sum(ctimestamp1)
+ round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 PREHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
 POSTHOOK: query: SELECT
- sum(ctimestamp1)
+ round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
 1123143.857
 PREHOOK: query: EXPLAIN SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 POSTHOOK: query: EXPLAIN SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 STAGE DEPENDENCIES:
@@ -837,7 +837,7 @@ STAGE PLANS:
                 outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7
                 Statistics: Num rows: 1 Data size: 64 Basic stats: COMPLETE Column stats: NONE
                 Select Operator
-                  expressions: _col0 (type: double), _col1 (type: double), _col2 (type: double), _col3 (type: double), _col4 (type: double), _col5 (type: double), _col6 (type: double), _col7 (type: double)
+                  expressions: round(_col0, 0) (type: double), _col1 BETWEEN 8.97077295279421E19 AND 8.97077295279422E19 (type: boolean), _col2 BETWEEN 8.97077295279421E19 AND 8.97077295279422E19 (type: boolean), _col3 BETWEEN 9.20684592523616E19 AND 9.20684592523617E19 (type: boolean), round(_col4, 3) (type: double), round(_col5, 3) (type: double), round(_col6, 3) (type: double), round(_col7, 3) (type: double)
                   outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7
                   Statistics: Num rows: 1 Data size: 64 Basic stats: COMPLETE Column stats: NONE
                   File Output Operator
@@ -855,29 +855,29 @@ STAGE PLANS:
         ListSink
 
 PREHOOK: query: SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 PREHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
 POSTHOOK: query: SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
-2.8798560435897438E13	8.970772952794215E19	8.970772952794215E19	9.206845925236167E19	9.471416447815086E9	9.471416447815086E9	9.471416447815086E9	9.595231068211004E9
+2.8798560435897E13	true	true	true	9.471416447815E9	9.471416447815E9	9.471416447815E9	9.595231068211E9
diff --git a/ql/src/test/results/clientpositive/tez/vectorized_timestamp_funcs.q.out b/ql/src/test/results/clientpositive/tez/vectorized_timestamp_funcs.q.out
index 4beabda..557d611 100644
--- a/ql/src/test/results/clientpositive/tez/vectorized_timestamp_funcs.q.out
+++ b/ql/src/test/results/clientpositive/tez/vectorized_timestamp_funcs.q.out
@@ -701,12 +701,12 @@ POSTHOOK: Input: default@alltypesorc_string
 1969-12-31 23:59:44.088	1970-01-01 00:00:15.007	39	40
 PREHOOK: query: -- SUM of timestamps are not vectorized reduce-side because they produce a double instead of a long (HIVE-8211)...
 EXPLAIN SELECT
-  sum(ctimestamp1)
+  round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 POSTHOOK: query: -- SUM of timestamps are not vectorized reduce-side because they produce a double instead of a long (HIVE-8211)...
 EXPLAIN SELECT
-  sum(ctimestamp1)
+  round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 STAGE DEPENDENCIES:
@@ -746,7 +746,7 @@ STAGE PLANS:
                 outputColumnNames: _col0
                 Statistics: Num rows: 1 Data size: 8 Basic stats: COMPLETE Column stats: NONE
                 Select Operator
-                  expressions: _col0 (type: double)
+                  expressions: round(_col0, 3) (type: double)
                   outputColumnNames: _col0
                   Statistics: Num rows: 1 Data size: 8 Basic stats: COMPLETE Column stats: NONE
                   File Output Operator
@@ -765,38 +765,38 @@ STAGE PLANS:
         ListSink
 
 PREHOOK: query: SELECT
- sum(ctimestamp1)
+ round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 PREHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
 POSTHOOK: query: SELECT
- sum(ctimestamp1)
+ round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
-1123143.8569999998
+1123143.857
 PREHOOK: query: EXPLAIN SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 POSTHOOK: query: EXPLAIN SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 STAGE DEPENDENCIES:
@@ -837,7 +837,7 @@ STAGE PLANS:
                 outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7
                 Statistics: Num rows: 1 Data size: 64 Basic stats: COMPLETE Column stats: NONE
                 Select Operator
-                  expressions: _col0 (type: double), _col1 (type: double), _col2 (type: double), _col3 (type: double), _col4 (type: double), _col5 (type: double), _col6 (type: double), _col7 (type: double)
+                  expressions: round(_col0, 0) (type: double), _col1 BETWEEN 8.97077295279421E19 AND 8.97077295279422E19 (type: boolean), _col2 BETWEEN 8.97077295279421E19 AND 8.97077295279422E19 (type: boolean), _col3 BETWEEN 9.20684592523616E19 AND 9.20684592523617E19 (type: boolean), round(_col4, 3) (type: double), round(_col5, 3) (type: double), round(_col6, 3) (type: double), round(_col7, 3) (type: double)
                   outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7
                   Statistics: Num rows: 1 Data size: 64 Basic stats: COMPLETE Column stats: NONE
                   File Output Operator
@@ -855,29 +855,29 @@ STAGE PLANS:
         ListSink
 
 PREHOOK: query: SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 PREHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
 POSTHOOK: query: SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
-2.8798560435897438E13	8.970772952794212E19	8.970772952794212E19	9.206845925236166E19	9.471416447815084E9	9.471416447815084E9	9.471416447815084E9	9.595231068211002E9
+2.8798560435897E13	true	true	true	9.471416447815E9	9.471416447815E9	9.471416447815E9	9.595231068211E9
diff --git a/ql/src/test/results/clientpositive/vectorized_timestamp_funcs.q.out b/ql/src/test/results/clientpositive/vectorized_timestamp_funcs.q.out
index 7a89552..f4e863f 100644
--- a/ql/src/test/results/clientpositive/vectorized_timestamp_funcs.q.out
+++ b/ql/src/test/results/clientpositive/vectorized_timestamp_funcs.q.out
@@ -666,12 +666,12 @@ POSTHOOK: Input: default@alltypesorc_string
 1969-12-31 23:59:44.088	1970-01-01 00:00:15.007	39	40
 PREHOOK: query: -- SUM of timestamps are not vectorized reduce-side because they produce a double instead of a long (HIVE-8211)...
 EXPLAIN SELECT
-  sum(ctimestamp1)
+  round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 POSTHOOK: query: -- SUM of timestamps are not vectorized reduce-side because they produce a double instead of a long (HIVE-8211)...
 EXPLAIN SELECT
-  sum(ctimestamp1)
+  round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 STAGE DEPENDENCIES:
@@ -705,7 +705,7 @@ STAGE PLANS:
           outputColumnNames: _col0
           Statistics: Num rows: 1 Data size: 8 Basic stats: COMPLETE Column stats: NONE
           Select Operator
-            expressions: _col0 (type: double)
+            expressions: round(_col0, 3) (type: double)
             outputColumnNames: _col0
             Statistics: Num rows: 1 Data size: 8 Basic stats: COMPLETE Column stats: NONE
             File Output Operator
@@ -723,38 +723,38 @@ STAGE PLANS:
         ListSink
 
 PREHOOK: query: SELECT
- sum(ctimestamp1)
+ round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 PREHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
 POSTHOOK: query: SELECT
- sum(ctimestamp1)
+ round(sum(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
-1123143.8569999998
+1123143.857
 PREHOOK: query: EXPLAIN SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 POSTHOOK: query: EXPLAIN SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 STAGE DEPENDENCIES:
@@ -789,7 +789,7 @@ STAGE PLANS:
           outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7
           Statistics: Num rows: 1 Data size: 64 Basic stats: COMPLETE Column stats: NONE
           Select Operator
-            expressions: _col0 (type: double), _col1 (type: double), _col2 (type: double), _col3 (type: double), _col4 (type: double), _col5 (type: double), _col6 (type: double), _col7 (type: double)
+            expressions: round(_col0, 0) (type: double), _col1 BETWEEN 8.97077295279421E19 AND 8.97077295279422E19 (type: boolean), _col2 BETWEEN 8.97077295279421E19 AND 8.97077295279422E19 (type: boolean), _col3 BETWEEN 9.20684592523616E19 AND 9.20684592523617E19 (type: boolean), round(_col4, 3) (type: double), round(_col5, 3) (type: double), round(_col6, 3) (type: double), round(_col7, 3) (type: double)
             outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7
             Statistics: Num rows: 1 Data size: 64 Basic stats: COMPLETE Column stats: NONE
             File Output Operator
@@ -807,29 +807,29 @@ STAGE PLANS:
         ListSink
 
 PREHOOK: query: SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 PREHOOK: type: QUERY
 PREHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
 POSTHOOK: query: SELECT
-  avg(ctimestamp1),
-  variance(ctimestamp1),
-  var_pop(ctimestamp1),
-  var_samp(ctimestamp1),
-  std(ctimestamp1),
-  stddev(ctimestamp1),
-  stddev_pop(ctimestamp1),
-  stddev_samp(ctimestamp1)
+  round(avg(ctimestamp1), 0),
+  variance(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_pop(ctimestamp1) between 8.97077295279421E19 and 8.97077295279422E19,
+  var_samp(ctimestamp1) between 9.20684592523616E19 and 9.20684592523617E19,
+  round(std(ctimestamp1), 3),
+  round(stddev(ctimestamp1), 3),
+  round(stddev_pop(ctimestamp1), 3),
+  round(stddev_samp(ctimestamp1), 3)
 FROM alltypesorc_string
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@alltypesorc_string
 #### A masked pattern was here ####
-2.8798560435897438E13	8.970772952794212E19	8.970772952794212E19	9.206845925236166E19	9.471416447815084E9	9.471416447815084E9	9.471416447815084E9	9.595231068211002E9
+2.8798560435897E13	true	true	true	9.471416447815E9	9.471416447815E9	9.471416447815E9	9.595231068211E9
-- 
1.7.9.5

